### DISCUSSION TITLE
[Question] How can I decrease the consumer lag for of Mirror Maker 2?

### DISCUSSION THREAD

**Discussion by msrijita18** [strimzi/strimzi-kafka-operator#7897] [Q&A]
When the kafka cluster is being pumped a heavy load of 40 millions messages of size 20 kb, the consumer lag for partitions are going as high as 150 K which takes about 1.5 hrs to catch up for the MM2.
I saw this [issue](https://github.com/strimzi/strimzi-kafka-operator/issues/2937) but its not that helpful.

**Comment by scholzj** [comment]
I guess sharing your current configuration might be a good start ;-)

**Comment by msrijita18** [comment]
I went through some resources, and found that making the no of replicas of MM2 equal to the no of partitions (=3) reduced the lag to as low as 0.

**Reply by scholzj** [reply]
Yeah, number of replicas and tasks was something I thought would be interesting to see from your configuration as playing with those might improve performance.

**Reply by msrijita18** [reply]
Yeah, these configs helped. Thanks.
